src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs

```diff
@@ -101,33 +99,21 @@ public static Scope CreateOutboundHttpScope(Tracer tracer, string httpMethod, Ur
                     return null;
                 }
 
-                string resourceUrl = requestUri != null ? UriHelpers.CleanUri(requestUri, removeScheme: true, tryRemoveIds: true) : null;
-                string httpUrl = requestUri != null ? UriHelpers.CleanUri(requestUri, removeScheme: false, tryRemoveIds: false) : null;
-
-                tags = new HttpTags();
-
-                string serviceName = tracer.Settings.GetServiceName(tracer, ServiceName);
-                scope = tracer.StartActiveWithTags(OperationName, tags: tags, serviceName: serviceName, spanId: spanId);
-
-                var span = scope.Span;
-
-                span.Type = SpanTypes.Http;
-                span.ResourceName = $"{httpMethod} {resourceUrl}";
-
-                tags.HttpMethod = httpMethod?.ToUpperInvariant();
-                tags.HttpUrl = httpUrl;
-                tags.InstrumentationName = IntegrationRegistry.GetName(integrationId);
-
-                tags.SetAnalyticsSampleRate(integrationId, tracer.Settings, enabledWithGlobalSetting: false);
+                var args = new OutboundHttpArgs
+                {
+                    SpanId = spanId,
+                    HttpMethod = httpMethod,
+                    IntegrationInfo = integrationId,
+                    RequestUri = requestUri,
+                };
+                var scope = tracer.OutboundHttpConvention.CreateScope(args, out tags);
```

Comment:
I always prefer to create a readonly struct in this case and if I have a lot of fields passing it by ref, something like [this](https://sharplab.io/#v2:EYLgHgbALANALiAlgGwD4AEBMBGAsAKHQGYACLEgYRIG8CT6zT0pKAnAUwEM52BlAYwD2AB3YVOyZO1YAKAJQ06DZQDdOrEuoDmAZxIBeEgDt2AdxIB5AK5xggq0YAmACThxhAQVa7F+Zf4ZeYU4jAElHAxJsTCIoAFYYJQDlV3cAWXY4AAtBCMMAIgBxAFEAFXzEv2TlACV2AEcrdh04AFVWREiTc3bEGXyst2EQAHoR/LlK6oBfAG4k/woObj4hURltHTl5qvpphYXEIx5WIwkybAAGNi4eARF2GWtbeydUz289TYVaXf90ADsmk+ADp3hlsrkdtUFvs/gsxiMSAAeHRWAC26PUAE8AHwIsYkLxaDHsY56Kw6dgRYDYlH8XH5ULPOwOFxDCiCIwqMlwRBc/LIkYMkEEpFCtGYnH4v5HE5nZAkFqsKz8OCWGyst5DYk6Ba/arEEhWZBcrQAfhIQRC4R2C2UiJRkqxrDx9oYjrqjWacAA5HpnKVSgAFEjozI5Ryiv4OwkSjEut0xhhG9BXEjgiNQgju+iO1EJ6W5kiehpNFr+kitGqhaPVfMjZ1F5P0I29Ehe8ttDrQhhw5QLVMsJa3dgAOTM91E4kk0nkvmqag0my6Zg7XEcXOQ2JZr3Z7l1MmLymisQSx4YRTKFQv9G6VY6/UG7lG4zkxe2xZHKwnpinjw4AAzYFdE/P5+wYQ5jmkBULmub8eF/f8ZCA9dOE3Ixt13Nl3l1ECtgXZJAXwsEhghSNez2A54T+fMm1dGVYyRYlSXJY0qRpOlkQZJlsO1dxOW5Xl+SMQVhVxOsPTjRtCwYqD5XOZYMO3JU4BVNU0OUndNT3XDPn1YsjSUrc6RNM1LWtMJHDtFsSzjeik3rQlOx9StAxDMMsyjYsGwcxiiNIYzMLpNNrkzSFrJzWy6NkxzklLb0Kz0atax86S/MMwKNxMh9Ohclpeko/xMs0ky+P3D5dBkMyjAtJVgismA4JIZ9hHI3ImvbDhEu7RAflvK0GvCSIdCGyLbJSMivMiVr2vG6pajLH120Mbqu0K4sIKo/ACGmIA=).

That's a personal preference non a blocker.

Reply:
Done https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/63/commits/317936c36374fdb9fe3691d0cd3273778ff5beb8

Reply:
Well, in big structures copying the whole struct can be more expensive than passing the struct by ref. But I agree in this case we don't need it, I did the test and the JIT is optimizing both calls in the same way: 

![image](https://user-images.githubusercontent.com/69803/109950269-a2264b00-7cdc-11eb-90ec-de0e253ceac7.png)